subsys/net/lib/shell/qbv_shell.c

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(net_shell);
+
+#include <zephyr/shell/shell.h>
+#include <stdlib.h>
+#include <string.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/net/ethernet.h>
+#include <zephyr/net/ethernet_mgmt.h>
+
+/* qbv enable <device> <value(off, on)> */
+static int cmd_net_qbv(const struct shell *sh, size_t argc, char **argv)
+{
+	shell_print(sh, "To set QBV config:");
+	shell_print(sh, "  1. Run enable to on");
+	shell_print(sh, "  2. Run set_config to set base_time/cycle_time/cycle_time_ext/list_len");
+	shell_print(sh, "  3. Run set_gc to set gate control");
+	shell_print(sh, "For example:");
+	shell_print(sh, "  1. net qbv enable switch_port@0 on");
+	shell_print(sh, "  2. net qbv set_config switch_port@0 200 0 0 10000000 0 2");
+	shell_print(sh, "  3. qbv set_gc switch_port@0 0 0x1 5000000");
+	shell_print(sh, "  4. qbv set_gc switch_port@0 0 0x2 5000000");
+
+	return 0;
+}
+
+static int parse_device_arg(const struct shell *sh, char **argv, const struct device **dev)
+{
+	*dev = shell_device_get_binding(argv[1]);
+	if (!*dev) {
+		shell_error(sh, "device %s not found", argv[1]);
+		return -ENODEV;
+	}
+	return 0;
+}
+
+/* qbv enable <device> <value(off, on)> */
+static int cmd_qbv_enable(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	struct net_if *iface;
+	struct ethernet_req_params params;
+	int ret;
+	bool enable;
+
+	ret = parse_device_arg(sh, argv, &dev);
+	if (ret < 0) {
+		return ret;
+	}
+
```

Comment:
For each command, we should probably verify that the interface is Ethernet type.
Why force user to use the device as a parameter if we are using interface internally? It should make things clearer if user just uses network interface index to configure Qbv.

Reply:
Please do not resolve the configuration but let the reviewer do it. See https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#workflow-suggestions-that-help-reviewers for details.

We should add here the interface type check, see my comment below.